README.md

```diff
@@ -4,221 +4,501 @@
+  <!-- [IntInfo] [c-no-update-expression-support] (24.9.19) See Bug IG-12510,
+    Requirement IG-5339, & DOC IG-12272. -->
+
+- <a id="method-write-kv-param-condition"></a>**condition** (Optional) (default: none) &mdash; a platform condition expression that defines conditions for performing the update.
+  For detailed information about platform condition expressions, see the [platform documentation](https://www.iguazio.com/docs/reference/latest-release/expressions/condition-expression/).
 
 Example:
 ```python
-data = [['tom', 10, 'TLV'], ['nick', 15, 'Berlin'], ['juli', 14, 'NY']] 
-df = pd.DataFrame(data, columns = ['name', 'age', 'city'])
-df.set_index('name')
-v3c.write(backend='kv', table='mytable', dfs=df, condition='age>14')
+data = [["tom", 10, "TLV"], ["nick", 15, "Berlin"], ["juli", 14, "NY"]]
+df = pd.DataFrame(data, columns = ["name", "age", "city"])
+df.set_index("name")
+v3c.write(backend="kv", table="mytable", dfs=df, condition="age>14")
 ```
 
-### Read
-Reads data from a backend.  
-Common read parameters:
-* iterator: bool - Return iterator of DataFrames or (if False) just one DataFrame
-* filter: string - Query filter (can't be used with query)
-* columns: []str - List of columns to pass (can't be used with query)
-* data_format: string - Data format (Not yet supported)
-* marker: string - Query marker (Not yet supported)
-* limit: int - Maximal number of rows to return (Not yet supported)
-* row_layout: bool - Weather to use row layout (vs the default column layout) (Not yet supported)
-
-
-#### backend specific parameters
-##### TSDB
-* start: string
-* end: string
-* step: string
-* aggregators: string
-* aggregationWindow: string
-* query: string - Query in SQL format
-* group_by: string - Query group by (can't be used with query)
-* multi_index: bool - Get the results as a multi index data frame where the labels are used as indexes
- in addition to the timestamp, or if `False` (default behavior) only the timestamp will function as the index.
-
-For detailed info on these parameters please visit [TSDB](https://github.com/v3io/v3io-tsdb#v3io-tsdb) docs.  
-Example:
+<a id="method-read"></a>
+### read Method
+
+Reads data from a data collection (table/stream) in a platform data container to a DataFrame according to the configured backend.
+
+- [Syntax](#method-read-syntax)
+- [Common parameters](#method-read-backend-common-params)
+- [`tsdb` backend `read` parameters](#method-read-params-tsdb)
+- [`kv` backend `read` parameters](#method-read-params-kv)
+- [`stream` backend `read` parameters](#method-read-params-stream)
+
+Reads data from a backend.
+
+<a id="method-read-syntax"></a>
+#### Syntax
+
 ```python
-df = client.read(backend='tsdb', query="select avg(cpu) as cpu, avg(diskio), avg(network)from mytable", start='now-1d', end='now', step='2h')
+read(backend=<backend>, table=<table>, attrs=<backend-specific parameters>)
```

Comment:
`attrs=<backend-specific parameters>` -> that's not really the syntax. The backends specific parameters are just written as normal parameters. See the examples you wrote below for reference.

Reply:
I fixed the syntax of all methods parameters; will be committed shortly to the PR branch + I added documentation of the common `write` `attrs` parameter and `execute` `args` parameter; will be committed shortly to the PR branch.